src/interfaces.ts

```diff
@@ -29,32 +37,76 @@ export interface AggregateVerifierLoginParams extends BaseLoginParams {
 export type LoginParams = SubVerifierDetailsParams | AggregateVerifierLoginParams;
 export type UserInfo = TorusVerifierResponse & LoginWindowResponse;
 
+export interface IdTokenLoginParams {
+  /**
+   * Name of the verifier created on Web3Auth Dashboard. In case of Aggregate Verifier, the name of the top level aggregate verifier.
+   */
+  verifier: string;
+
+  /**
+   * Unique Identifier for the User. The verifier identifier field set for the verifier/ sub verifier. E.g. "sub" field in your on jwt id token.
+   */
+  verifierId: string;
+
+  /**
+   * The idToken received from the Auth Provider.
+   */
+  idToken: string;
+
+  /**
+   * Name of the sub verifier in case of aggregate verifier setup. This field should only be provided in case of an aggregate verifier.
+   */
+  subVerifier?: string;
+
+  /**
+   * Extra verifier params in case of a WebAuthn verifier type.
+   */
+  extraVerifierParams?: WebAuthnExtraParams;
+
+  /**
+   * Any additional parameter (key value pair) you'd like to pass to the login function.
+   */
+  additionalParams?: ExtraParams;
+}
+
 // TODO extend interface and type documentation wherever necessary. @Yash
 export interface ICoreKit {
-  /** The tKey instance, if initialized. */
+  /** The tKey SDK instance
+   *  TKey is the core module on which this wrapper SDK sits for easy integration.
+   **/
   tKey: ThresholdKey | undefined;
 
   // TODO document errors across all interface methods! maybe even define error
   // codes and document which are thrown. in particular here, error is thrown if
   // not factor key is given (either as function parameter or through local
   // storage)
-  // TODO login with id token @Yash
   /**
-   * Login to tKey and initialize all relevant components.
-   * @param loginParams - TKey login parameters.
+   * Login into the SDK in an implicit flow and initialize all relevant components.
+   * @param loginParams - Parameters for Implicit Login.
+   * @param factorKey - A BN used for encrypting your Device/ Recovery TSS Key Share. Optional for new users, mandatory for existing users, if not provided we will try to fetch it from local storage. Read more about it here: https://web3auth.io/docs/infrastructure/mpc-architecture#factor-keys
```

Comment:
I suggest we don't include links to docs here, as these can easily break in the future. The developer can always find the docs via website.